internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -986,19 +986,47 @@ func (r *ClusterDeploymentsReconciler) syncManifests(ctx context.Context, log lo
 }
 
 func (r *ClusterDeploymentsReconciler) getClusterDeploymentManifest(ctx context.Context, log logrus.FieldLogger, clusterInstall *hiveext.AgentClusterInstall) (map[string]string, error) {
-	configuredManifests := &corev1.ConfigMap{}
-	configuredManifests.Data = map[string]string{}
-	// get manifests from configmap if we have reference for it
-	if clusterInstall.Spec.ManifestsConfigMapRef != nil {
-		err := r.Get(ctx, types.NamespacedName{Namespace: clusterInstall.Namespace,
-			Name: clusterInstall.Spec.ManifestsConfigMapRef.Name}, configuredManifests)
+	configuredManifests := map[string]string{}
+
+	// Get manifests from referenced ConfigMaps (fallback to single ManifestsConfigMapRef)
+	if clusterInstall.Spec.ManifestsConfigMapRefs != nil {
+		for _, ref := range clusterInstall.Spec.ManifestsConfigMapRefs {
+			configMap, err := r.getManifestConfigMap(ctx, log, clusterInstall, ref.Name)
+			if err != nil {
+				return nil, err
+			}
+			// Add data to manifests map
+			for k, v := range configMap.Data {
+				configuredManifests[k] = v
```

Comment:
Should there be enforcement for that? Or at least a warning?
```suggestion
				if _, exists := configuredManifests[k]; exists {
					// fail or log a warning here
				}
				configuredManifests[k] = v
```

Reply:
The way the code is structured, the effect will be to ignore all but the last key with the same name. I'm just concerned this could result in some pretty weird behavior that the user would have a hard time understanding. If you assert it shouldn't be a hard failure, okay :man_shrugging: -- but yeah, at least a warning in the logs to acknowledge that we checked and are aware we're doing something odd.

Reply:
looks like a product decision, in the rest-api we enable to override exiting manifests, for example if user need to update them. 
here the case is a bit different so maybe we can fail by protecting the user. if it will be a log it will help us only in the debug flow. 
@romfreiman @avishayt what do you think?

Reply:
Is there a good reason not to fail in this case?

Reply:
> Is there a good reason not to fail in this case?

It's a user configuration issue that isn't (necessarily) harmful. So not sure we need hard failure here:/

Reply:
Would a user ever do this on purpose?  I think it would always indicate a mistake that we should validate (just like we validate other bad inputs).

Reply:
mco takes the highest number on the file if Im not wrong. this is how they merge collisions. Is it the concern here? to confignames with same name for example?

Reply:
> Is it the concern here? to confignames with same name for example?

Since they have to be in the same namespace, we don't have to worry about the names of the _ConfigMaps_ colliding. What I'm concerned about is the user (not reading/understanding the API doc -- a common occurrence I'm sure we can all agree -- and) specifying multiple ConfigMaps with colliding data keys. A contrived example:

```yaml
---
kind: ConfigMap
metadata:
  name: rbac-for-foo
data:
  role:
    ...
  rolebinding:
    ...
---
kind: ConfigMap
metadata:
  name: rbac-for-bar
data:
  role:
    ...
  rolebinding:
    ...
```

...and

```yaml
...
manifestConfigMapRefs:
- rbac-for-foo
- rbac-for-bar
```
In this case we would lay down the `rbac-for-bar` role and rolebinding, and ignore `rbac-for-foo`.

Just a thought: I understand that the keys end up mapping to filenames; but is there a significance to those filenames? What if instead of using the data key as the map key (and thus filename), we used a unique string instead? A UUID would work; but for traceability what about joining the name of the original configmap with the data key?

```suggestion
				configuredManifests[fmt.Sprintf("%s.%s", configMap.Name, k)] = v
```

This way we don't have the uniqueness restriction at all.

Reply:
Like this: https://github.com/openshift/assisted-service/pull/3400

Reply:
I would rather not change customers filesnames since they expected to debug if there is an issue, right?

Reply:
Do they? I don't know. In any case, it should be pretty obvious what came from where, especially if we document the naming scheme.

Back to enforcing uniqueness, I see we already throw errors for various problems with the customer-provided manifests, including:
- [Keys with path separators](https://github.com/danielerez/assisted-service/blob/6d56da0f48212b8c3dbd424e006c653f04c904a4/internal/manifests/manifests.go#L73)
- [Non/improperly base64-encoded content](https://github.com/danielerez/assisted-service/blob/6d56da0f48212b8c3dbd424e006c653f04c904a4/internal/manifests/manifests.go#L78)
- Invalid [yaml](https://github.com/danielerez/assisted-service/blob/6d56da0f48212b8c3dbd424e006c653f04c904a4/internal/manifests/manifests.go#L87) or [json](https://github.com/danielerez/assisted-service/blob/6d56da0f48212b8c3dbd424e006c653f04c904a4/internal/manifests/manifests.go#L91) formatting
- [Files of types other than yaml or json](https://github.com/danielerez/assisted-service/blob/6d56da0f48212b8c3dbd424e006c653f04c904a4/internal/manifests/manifests.go#L95)

So why shouldn't we throw an error for duplicated filenames across configmaps?

Reply:
Right, so we can go with conflict validation for now, pushed: https://github.com/openshift/assisted-service/pull/3429

Reply:
Resolved via https://github.com/openshift/assisted-service/pull/3429